### ISSUE TITLE
fatal error: driver/twai.h: No such file or directory

### ISSUE THREAD

**Issue by jdrescher2006** [mb-software/esphome-huawei-r4850#5]
I am having trouble compiling this because of this line `#include <driver/twai.h>` in the esp32_can.cpp file. 
It appears that the driver comes from Espressif IoT Development Framework: [https://github.com/espressif/esp-idf](https://github.com/espressif/esp-idf).
On my linux (Ubuntu) system I tried to set it up after this instruction: [https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/linux-macos-setup.html](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/linux-macos-setup.html).

The instruction says you have to run the command `. $HOME/esp/esp-idf/export.sh` in the folder where you want to use it.
Unfortunately it still does not work: `src/esphome/components/esp32_can/esp32_can.cpp:5:25: fatal error: driver/twai.h: No such file or directory`
How did you get this to work?

**Comment by jdrescher2006** [comment]
So I learned that twai is part of the esp-idf framework. So I set the framework to esp-idf. Problem is: does not work.
Reason is that multiple components such as ethernet or web_server are not supported by esp-idf and will work only with arduino framework. Also if I remove those, it does not compile on my linux system. Nice.
So I am stuck with arduino framework, but that does not support twai driver. Next thing to try is import all the C files manually. This will be a lot of fun :-(
